### PR TITLE
Internal: Documentation of `document/elements` component. [ED-6167]

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+    const eSection = $e.run( 'document/elements/create', {
+      container: elementor.getPreviewContainer(),
+      columns: 1,
+      model: { elType: 'section' },
+    } );  
+    ```
+    Result: ![empty-column](./_images/empty-column.png)
+    Now to create widget, we need column, here is two examples how to reach column:
+    ```javascript
+    const eColumn = eSection.view.children.findByIndex( 0 ).getContainer();
```
Comment (iNewLegend) [inline docs/assets/dev/js/editor/document/elements/readme.md:77]:
```suggestion
    const eColumn = eSection.children[0]
```

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 2]
```diff
@@ -0,0 +1,313 @@
+
+## Component -- `$e.components.get('document/elements')`
+
+*  **Name**: Elements.
+*  **Description**: Provide a way to manage elements, create, edit, delete. copy, paste, etc...
```
Comment (iNewLegend) [inline docs/assets/dev/js/editor/document/elements/readme.md:5]:
```suggestion
*  **Description**: Provide a way to  manipulate the document elements: create, edit, delete. copy, paste, etc...
```

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+    | _model_      | `{Object}`       | Model to create.
+    | _options_    | `{Object}`       | Options.
+
+    **_options:_**
+   
+    | Property    | Type                              | Default   | Description                            |
+    |-------------|-----------------------------------|-----------|----------------------------------------|
+    | at          | `{Number}`                        | `{null}`  | Position (null means last). 
+    | clone       | `{Boolean}`                       | `{false}` | Generate unique id for the model.
+    | edit        | `{Boolean}`                       | `{false}` | Is turn edit panel for the new element.
```
Comment (StyleShit) [inline docs/assets/dev/js/editor/document/elements/readme.md:59]:
```suggestion
    | edit        | `{Boolean}`                       | `{false}` | Whether to open the panel for editing after creating the element.
```

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+    | _container_  | `{Container}`    | Target.
+    | _containers_ | `{Container[]}`  | Targets.
+    | _model_      | `{Object}`       | Model to create.
+    | _options_    | `{Object}`       | Options.
+
+    **_options:_**
+   
+    | Property    | Type                              | Default   | Description                            |
+    |-------------|-----------------------------------|-----------|----------------------------------------|
+    | at          | `{Number}`                        | `{null}`  | Position (null means last).
```
Comment (StyleShit) [inline docs/assets/dev/js/editor/document/elements/readme.md:57]: Maybe merge with the previous table, and do like `options.at`, etc ?

Comment (StyleShit) [inline docs/assets/dev/js/editor/document/elements/readme.md:57]:
```suggestion
    | at          | `{Number}`                        | `{null}`  | Position (`null` means last). 
```

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+    const eSection = $e.run( 'document/elements/create', {
+      container: elementor.getPreviewContainer(),
+      columns: 1,
+      model: { elType: 'section' },
+    } );  
+    ```
+    Result: ![empty-column](./_images/empty-column.png)
+    Now to create widget, we need column, here is two examples how to reach column:
+    ```javascript
+    const eColumn = eSection.children[0];
```
Comment (StyleShit) [inline docs/assets/dev/js/editor/document/elements/readme.md:77]:
```suggestion
    const eColumn = eSection.children[ 0 ];
```

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+    ```javascript
+    // Create section.
+    const eSection = $e.run( 'document/elements/create', {
+      container: elementor.getPreviewContainer(),
+      columns: 1,
+      model: { elType: 'section' },
+    } );  
+    ```
+    Result: ![empty-column](./_images/empty-column.png)
+    Now to create widget, we need column, here is two examples how to reach column:
```
Comment (StyleShit) [inline docs/assets/dev/js/editor/document/elements/readme.md:75]:
```suggestion
    Now, in order to create the widget, we need a column. There are two ways to access the column:
```

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+    ```javascript
+    const eColumn = eSection.children[0];
+    ```
+    OR
+    ```javascript
+    const eColumn = elementor.getContainer('cb70e3c');
+    ```
+    Then let create heading widget at the column we reach above:
```
Comment (StyleShit) [inline docs/assets/dev/js/editor/document/elements/readme.md:83]:
```suggestion
    Finally, the heading widget can be created at the selected column:
```

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+   |---           |---               |---|
+   | _container_  | `{Container}`    | Target.
+   | _containers_ | `{Container[]}`  | Targets.
+    
+* **Examples**:
+    Delete a section, Assuming we have this section, and want to delete it.
+    ![widget-heading](./_images/widget-heading.png)
+    ```javascript
+    // Get section container.
+    const eSection = elementor.getContainer('886643f');
```
Comment (StyleShit) [inline docs/assets/dev/js/editor/document/elements/readme.md:111]:
```suggestion
    const eSection = elementor.getContainer( '886643f' );
```

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 9]
```diff
@@ -0,0 +1,313 @@
...
+    
+    $e.run( 'document/elements/delete', { 
+        container: eSection,
+    } );
+    ```
+    result: section deleted and all children elements.
```
Comment (StyleShit) [inline docs/assets/dev/js/editor/document/elements/readme.md:117]:
```suggestion
    Result: The section and all of its child elements are deleted.
```

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 10]
```diff
@@ -0,0 +1,313 @@
...
+*  **Returns**: `{Container | Container[]}` *Created container(s)*.
+*  **Arguments**: 
+
+   | Property     | Type             | Description |
+   |---           |---               |---|
+   | _container_  | `{Container}`    | Target.
+   | _containers_ | `{Container[]}`  | Targets.
+    
+* **Examples**:
+    Duplicate a section, Assuming we have this section, and want to duplicate it.
```
Comment (StyleShit) [inline docs/assets/dev/js/editor/document/elements/readme.md:131]:
```suggestion
    Duplicating a section.
```

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 11]
```diff
@@ -0,0 +1,313 @@
...
+    ![widget-heading](./_images/widget-heading.png)
+    ```javascript
+    // Get section container.
+    const eSection = elementor.getContainer('886643f');
```
Comment (StyleShit) [inline docs/assets/dev/js/editor/document/elements/readme.md:135]:
```suggestion
    const eSection = elementor.getContainer( '886643f' );
```

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 12]
```diff
@@ -0,0 +1,313 @@
...
+   
+    | Property         | Type                              | Default   | Description                            |
+    |------------------|-----------------------------------|-----------|----------------------------------------|
+    | withPageSettings | `{Boolean}`                       | `{false}` | Should set page settings of `args.page_settings`.
+    | at               | `{Number}`                        | `{null}`  | Position ( Automatically increased ).
+    | clone            | `{Boolean}`                       | `{false}` | Generate unique id for the model.
+    | edit             | `{Boolean}`                       | `{false}` | Is turn edit panel for the new element.
+    | onBeforeAdd      | `{function()}`                    |           | Run callback before add.
+    | onAfterAdd       | `{function( newModel, newView )}` |           | Run callback after add.
+    | trigger          | `{Boolean}`                       | `{false}` | *Deprecated*.
```
Comment (StyleShit) [inline docs/assets/dev/js/editor/document/elements/readme.md:183]: Like before, maybe merge with the table above.

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 13]
```diff
@@ -0,0 +1,313 @@
...
+    | page_settings | `{Object}` |           | Template page settings.
+    
+    **_options:_**
+   
+    | Property         | Type                              | Default   | Description                            |
+    |------------------|-----------------------------------|-----------|----------------------------------------|
+    | withPageSettings | `{Boolean}`                       | `{false}` | Should set page settings of `args.page_settings`.
+    | at               | `{Number}`                        | `{null}`  | Position ( Automatically increased ).
+    | clone            | `{Boolean}`                       | `{false}` | Generate unique id for the model.
+    | edit             | `{Boolean}`                       | `{false}` | Is turn edit panel for the new element.
```
Comment (StyleShit) [inline docs/assets/dev/js/editor/document/elements/readme.md:180]: Same as above

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 14]
```diff
@@ -0,0 +1,313 @@
...
+    |---------------|------------|-----------|----------------------------------------|
+    | content       | `{Object}` |           | Template content. 
+    | page_settings | `{Object}` |           | Template page settings.
+    
+    **_options:_**
+   
+    | Property         | Type                              | Default   | Description                            |
+    |------------------|-----------------------------------|-----------|----------------------------------------|
+    | withPageSettings | `{Boolean}`                       | `{false}` | Should set page settings of `args.page_settings`.
+    | at               | `{Number}`                        | `{null}`  | Position ( Automatically increased ).
```
Comment (StyleShit) [inline docs/assets/dev/js/editor/document/elements/readme.md:178]:
```suggestion
    | at               | `{Number}`                        | `{null}`  | Position (Automatically increased).
```

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 15]
```diff
@@ -0,0 +1,313 @@
...
+    const eColumn = elementor.getContainer('cb70e3c'),
+        eWidget = elementor.getContainer('2e4b783');
+    
+    $e.run('document/elements/copy', { 
+        container: eWidget
+    } );
+    
+    const pastedContainers = $e.run('document/elements/paste', { 
+        container: eColumn
+    } );
```
Comment (StyleShit) [inline docs/assets/dev/js/editor/document/elements/readme.md:210]:
```suggestion
    const eColumn = elementor.getContainer( 'cb70e3c' ),
        eWidget = elementor.getContainer( '2e4b783' );
    
    $e.run( 'document/elements/copy', { 
        container: eWidget
    } );
    
    const pastedContainers = $e.run( 'document/elements/paste', { 
        container: eColumn
    } );
```

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 16]
```diff
@@ -0,0 +1,313 @@
...
+   | _settings_        | `{Object}`  | Settings.
+   | _isMultiSettings_ | `{Boolean}` | default: `{false}`, settings for multi containers (each container).
+   | _options_         | `{Object}`  | 
+
+    **_options:_**
+   
+    | Property         | Type        | Default   | Description                            |
+    |------------------|------------ |-----------|----------------------------------------|
+    | external         | `{Boolean}` | `{false}` | Re-render panel with new settings.
+    | debounce         | `{Boolean}` | `{true}`  | Enable\Disable debounce.
```
Comment (StyleShit) [inline docs/assets/dev/js/editor/document/elements/readme.md:294]:
```suggestion
    | debounce         | `{Boolean}` | `{true}`  | Enable/Disable debounce.
```

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 17]
```diff
@@ -0,0 +1,313 @@
+
+## Component -- `$e.components.get('document/elements')`
+
+*  **Name**: Elements.
+*  **Description**: Provide a way to  manipulate the document elements: create, edit, delete. copy, paste, etc...
+
+## Component `document/elements/` -- Commands
```
[the remark this answers is not in the document]
Comment (iNewLegend) [reply docs/assets/dev/js/editor/document/elements/readme.md:7]: https://github.com/elementor/elementor/blob/a13152377c4124f7f453be6fa8aa183fc9be22b2/docs/assets/dev/js/editor/document/dynamic/readme.md
We have to follow the format.

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 18]
```diff
@@ -0,0 +1,313 @@
...
+    | _options_    | `{Object}`       | Options.
+
+    **_options:_**
+   
+    | Property    | Type                              | Default   | Description                            |
+    |-------------|-----------------------------------|-----------|----------------------------------------|
+    | at          | `{Number}`                        | `{null}`  | Position (null means last). 
+    | clone       | `{Boolean}`                       | `{false}` | Generate unique id for the model.
+    | edit        | `{Boolean}`                       | `{false}` | Is turn edit panel for the new element.
+    | onBeforeAdd | `{function()}`                    |           | Run callback before add.
```
[the remark this answers is not in the document]
Comment (iNewLegend) [reply docs/assets/dev/js/editor/document/elements/readme.md:60]: It's the way JSDOC describe a callback.

File: docs/assets/dev/js/editor/document/elements/readme.md (hunk 12 above)
Comment (iNewLegend) [reply docs/assets/dev/js/editor/document/elements/readme.md]: Separation make it more understand able.

File: docs/assets/dev/js/editor/document/elements/readme.md (hunk 13 above)
Comment (iNewLegend) [reply docs/assets/dev/js/editor/document/elements/readme.md:180]: Separation make it more understand able.

**Review by StyleShit — APPROVED** [review]

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 19]
```diff
@@ -0,0 +1,313 @@
...
+    | at               | `{Number}`                        | `{null}`  | Position (Automatically increased).
+    | clone            | `{Boolean}`                       | `{false}` | Generate unique id for the model.
+    | edit             | `{Boolean}`                       | `{false}` | Is turn edit panel for the new element.
+    | onBeforeAdd      | `{function()}`                    |           | Run callback before add.
+    | onAfterAdd       | `{function( newModel, newView )}` |           | Run callback after add.
+    | trigger          | `{Boolean}`                       | `{false}` | *Deprecated*.
+
+## Paste _Command_ -- `$e.run('document/elements/paste')`
+*  **Name**: Paste.
+*  **Description**: Paste to container.
```
Comment (ItzhakAvm) [inline docs/assets/dev/js/editor/document/elements/readme.md:187]:
```suggestion
*  **Description**: Paste new element into an existing element.
```

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 20]
```diff
@@ -0,0 +1,313 @@
...
+        container: eColumn
+    } );
+    
+    console.log( pastedContainers );
+    ```
+    Result:![widget-heading-pasted](./_images/widget-heading-pasted.png)
+
+## Paste-Style _Command_ -- `$e.run('document/elements/paste-style')`
+*  **Name**: Paste-Style.
+*  **Description**: Paste style to container.
```
Comment (ItzhakAvm) [inline docs/assets/dev/js/editor/document/elements/readme.md:218]:
```suggestion
*  **Description**: Paste style to element.
```

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 21]
```diff
@@ -0,0 +1,313 @@
...
+    
+    $e.run('document/elements/paste-style', { 
+        container: eSpecialWidget
+    } );
+    ```
+    Result:![widget-heading-pasted](./_images/widget-heading-pasted.png)
+
+## Reset-Style _Command_ -- `$e.run('document/elements/reset-style')`
+*  **Name**: Reset-Style.
+*  **Description**: Rest style of container.
```
Comment (ItzhakAvm) [inline docs/assets/dev/js/editor/document/elements/readme.md:249]:
```suggestion
*  **Description**: Rest style of an element.
```

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 22]
```diff
@@ -0,0 +1,313 @@
...
+      container: eWidget
+    } );
+    ```
+    Result:
+    
+    ![widget-heading-pasted](./_images/widget-heading-pasted.png)
+
+## Settings _Command_ -- `$e.run('document/elements/settings')`
+*  **Name**: Settings.
+*  **Description**: Change container settings.
```
Comment (ItzhakAvm) [inline docs/assets/dev/js/editor/document/elements/readme.md:277]:
```suggestion
*  **Description**: Change element settings.
```

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 23]
```diff
@@ -0,0 +1,313 @@
...
+    | _container_  | `{Container}`    | Target.
+    | _containers_ | `{Container[]}`  | Targets.
+    | _model_      | `{Object}`       | Model to create.
+    | _options_    | `{Object}`       | Options.
+
+    **_options:_**
+   
+    | Property    | Type                              | Default   | Description                            |
+    |-------------|-----------------------------------|-----------|----------------------------------------|
+    | at          | `{Number}`                        | `{null}`  | Position (`null` means last).
```
Comment (ItzhakAvm) [inline docs/assets/dev/js/editor/document/elements/readme.md:57]:
```suggestion
    | at          | `{Number}`                        | `{null}`  | Index position (`null` means last). 
```

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 24]
```diff
@@ -0,0 +1,313 @@
...
+    ```javascript
+    // Create section.
+    const eSection = $e.run( 'document/elements/create', {
+      container: elementor.getPreviewContainer(),
+      columns: 1,
+      model: { elType: 'section' },
+    } );  
+    ```
+    Result: ![empty-column](./_images/empty-column.png)
+    Now, in order to create the widget, we need a column. There are two ways to access the column:
```
Comment (ItzhakAvm) [inline docs/assets/dev/js/editor/document/elements/readme.md:75]:
```suggestion
    Now, in order to create the widget, we need a column. There are two ways to access the automatically created column, depends on the context. When the section's container instance is available, we can simply access its children:
```

File: docs/assets/dev/js/editor/document/elements/readme.md [hunk 25]
```diff
@@ -0,0 +1,313 @@
...
+    ```javascript
+    const eColumn = eSection.children[ 0 ];
+    ```
+    OR
```
Comment (ItzhakAvm) [inline docs/assets/dev/js/editor/document/elements/readme.md:79]:
```suggestion
    In other cases, where a the column's container-id is provided, we can retrieve the corresponding container instance:
```

**Review by YitzhakAvm — CHANGES_REQUESTED** [review]
Good job 👍🏻
It can be a nice addition to describe select/deselect commands, and other omitted ones - for completeness.

**Review by YitzhakAvm — APPROVED** [review]

**Pull request merged**